src/factory.js

```diff
@@ -131,6 +134,11 @@ module.exports = ({ Component, createElement, PropTypes }) => {
                       completeStateKeys,
```

Comment:
We could remove `k` from the spy because there is a special behavior on this, it will allow us to remove the `configurable` above.

Reply:
Do you mean like this : 
```js
                    const _completeStateKeys = { ...completeStateKeys }
                    delete _completeStateKeys[k];
                    [propsProxy, propsSpy] = makeSpy(propsKeys, propsAccessor);
                    [stateProxy, stateSpy] = makeSpy(
                      _completeStateKeys,
                      stateAccessor
                    )
                    // Removed code
                    // Object.defineProperty(stateProxy, k, {
                    //   enumerable: false,
                    //   get () {
                    //     throw new CircularComputedError(k)
                    //   },
                    // })
```

Reply:
No.

What the type of `completeStateKeys`?

Reply:
my bad, it's an array, So normally when I remove the key from the `completeStateKeys` we will not need any extra code and we can put the as not configurable

Reply:
```js
const computedIndex = completeStateKeys.indexOf(k)
                    if (computedIndex > -1) {
                      completeStateKeys.splice(computedIndex, 1)
                    }
                    [propsProxy, propsSpy] = makeSpy(propsKeys, propsAccessor);
                    [stateProxy, stateSpy] = makeSpy(
                      _completeStateKeys,
                      stateAccessor
                    )
                    // Removed code
                    // Object.defineProperty(stateProxy, k, {
                    //   enumerable: false,
                    //   get () {
                    //     throw new CircularComputedError(k)
                    //   },
                    // })
```

Reply:
Nope, this will break everything.

You should be able to figure out why.

Reply:
My bad, I meant to splice the copied array and not main one

```js
                   const _completeStateKeys = completeStateKeys
                    const computedIndex = completeStateKeys.indexOf(k)
                    if (computedIndex > -1) {
                      _completeStateKeys.splice(computedIndex, 1)
                    }
                    [propsProxy, propsSpy] = makeSpy(propsKeys, propsAccessor);
                    [stateProxy, stateSpy] = makeSpy(
                      _completeStateKeys,
                      stateAccessor
                    )
```

Reply:
Why so complicated instead of simply using [`Array#filter`](https://devdocs.io/javascript/global_objects/array/filter)?

Reply:
I don't get the idea for this comment https://github.com/JsCommunity/reaclette/pull/23#discussion_r261967165
By removing K from the spy, we remove tracibility change for the computed named `K` and it becomes undefined.
Why do you want to do that ?

Reply:
We don't need to spy the property `k` because it will not be accessed (should not) from the computed of the same name.

That's exactly what you are working on in this PR.

Reply:
I understand, but making `k` not accessible inside its computed makes things unclear for the developer who calls a computed in it self, don't you think so ?